packages/core/src/overlay/OverlayHeader.tsx

```diff
@@ -0,0 +1,75 @@
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import clsx from "clsx";
+import {
+  type ComponentPropsWithoutRef,
+  type ReactNode,
+  forwardRef,
+} from "react";
+import { H4, Text } from "../text";
+import { makePrefixer } from "../utils";
+import overlayHeaderCss from "./OverlayHeader.css";
+
+const withBaseName = makePrefixer("saltOverlayHeader");
+
+export interface OverlayPanelContentProps
+  extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * Description text is displayed just below the header
+   **/
+  description?: ReactNode;
+  /**
+   * End adornment component
+   */
+  actions?: ReactNode;
```

Comment:
https://github.com/jpmorganchase/salt-ds/pull/4138#discussion_r1766840122

updated endAdornment for actions

```suggestion
  /**
   * Actions to be displayed in header
   */
  actions?: ReactNode;
```